evap/student/templates/student_vote.html

```diff
@@ -300,16 +299,33 @@ <h3 class="mb-3">{{ evaluation.full_name }} ({{ evaluation.course.semester.name
             });
 
             document.querySelectorAll("[data-mark-no-answers-for]").forEach(button => {
+                const contributorId = button.dataset.markNoAnswersFor;
+                const voteArea = document.querySelector(`#vote-area-${contributorId}`);
+                const collapseToggle = voteArea.closest(".collapsible").querySelector(".collapse-toggle");
+
                 button.addEventListener("click", () => {
-                    const contributorId = button.dataset.markNoAnswersFor;
-                    const voteArea = document.querySelector(`#vote-area-${contributorId}`);
                     // check "no answer" and uncheck all other options
                     voteArea.querySelectorAll(".vote-inputs [type=radio][value='6']")
                         .forEach(radioInput => radioInput.click());
 
                     // hide questionnaire for contributor
                     $(voteArea).collapse("hide");
+                    collapseToggle.classList.add("tab-selectable");
+
+                    // Disable this button, until user changes a value
+                    button.classList.remove("tab-selectable");
+                    button.disabled = true;
```

Comment:
:bulb: I like the feature that the button gets disabled once you activate it. But it would be even nicer if the vote area gets collapsed and the button disabled if all vote buttons were “no answer” beforehand. If this idea doesn’t fit in here nicely, we can open an issue afterwards (don’t want to do unnecessary scope creeping here)

Reply:
I think it wouldn't be too hard to implement this "hook" that happens on every press of a "no-answer" button and checks if all buttons are "no-answer". The site should also scroll to the contributor name afterwards, so that you wouldn't have to scroll up again.

I don't know how well this would integrate with the upcoming "additional-text-answers-to-all-questions" (naming is hard) feature, which is why I would leave this to wait until after both of the pull requests are merged to come up with something elegant then.

I don't know if this would be widely used though... I mean do you ever go through a contributor and think "hm, no, I really cannot answer this particular question, but maybe something else comes up" at every question? (or is there another use case?)

Maybe a "mark all others as no-answer and collapse" button? Or maybe the collapse button thingy should stick to the top of the main container so that you could collapse the contributor after you are done?

Reply:
Ok, let me rephrase my idea: When clicking »I can’t give feedback for this contributor« and then reloading the page, the card will reappear uncollapsed with all buttons at “no answer”. It is only about the persistence of the ui when the page reloads, sorry for not delivering this keyword to you :)

I think the phrasing of this button is well designed, I think the main use cases include: someone does not know the person’s name, wasn’t in contact with the contributor (like a tutor of another group), or is indifferent about this contributor. But the button could have some unexpected behavior if the voter answers some questions first and then clicks the button and – woopsie – all answers for this contributor were reset. But those advanced thoughts are definitely outside of this scope :)

Reply:
Oh, that makes sense. Honestly, let's just open another issue afterwards like you suggested from the start :sweat_smile: